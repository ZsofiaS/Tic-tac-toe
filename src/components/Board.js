import React from 'react';
import styled from 'styled-components';

const BoardContainer = styled.div`
    width: 630px;
    height: 630px;
    padding: 1rem;
    border: 1px solid gainsboro;
    margin: 2rem auto;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
`;

const Square = styled.button`
    width: 200px;
    height: 200px;
    border: 1px solid crimson;
    margin: 4px;
`;

const Board = () => {
    return (
        <BoardContainer>
            <Square value="1" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="2" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="3" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="4" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="5" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="6" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="7" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="8" onClick={(e) => console.log(e.target.value)}></Square>
            <Square value="9" onClick={(e) => console.log(e.target.value)}></Square>
        </BoardContainer>
    )
}

export default Board;