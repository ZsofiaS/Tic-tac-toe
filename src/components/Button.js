import styled, { css } from 'styled-components';

const Button = styled.button`
  cursor: pointer;
  background: transparent;
  border-radius: 3px;
  color: black;
  border: 2px solid;
  border-image-slice: 1;
  border-image-source: linear-gradient(to left, #118ab2, #06d6a0, #ffd166);
  background-image: linear-gradient(to left, #118ab2, #06d6a0, #ffd166);
  margin: 0 1em;
  padding: 0.5em 0.8em;
  transition: 0.3s all ease-out;
  font-family: 'Grandstander', cursive;
  font-size: 18px;
  color: white;

  // import css from styled components & interpolate a function into the template literal
  // the function gets passed the props of our component
  ${(props) =>
    props.primary &&
    css`
      color: crimson;
      border: 2px solid crimson;
    `};

  &:hover {
    background-color: none;
    background-image: none;
    color: black;
  }
`;

export default Button;
