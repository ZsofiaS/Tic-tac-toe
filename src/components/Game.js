import React, { useState } from 'react';
import { Board } from './Board';

const Game = () => {
  const [currentPlayer, setCurrentPlayer] = useState('X');
  const [winner, setWinner] = useState('');

  const updatePlayer = () => {
    currentPlayer === 'X' ? setCurrentPlayer('O') : setCurrentPlayer('X');
  };

  const declareWinner = () => {
    currentPlayer === 'X' ? setWinner('O') : setWinner('X');
  };

  const restartGame = () => {
      setWinner('');
      setCurrentPlayer('X');
  }

  return (
    <>
      {winner !== '' ? <p>Winner: {winner}</p> : <p>Player: {currentPlayer}</p>}
      <Board
        current={currentPlayer}
        updateHandler={updatePlayer}
        declareWinner={declareWinner}
        restart={restartGame}
      />
    </>
  );
};

export default Game;
