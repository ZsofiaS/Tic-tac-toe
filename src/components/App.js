import React from 'react';
import styled from 'styled-components';
import Button from './Button';
import Game from './Game';

const Container = styled.div`
  text-align: center;
  padding: 1rem;
`;

function App() {
  return (
    <Container>
      <h1>Tic-tac-toe</h1>
      <p>Styled components & React Hooks</p>
      <Button onClick={() => console.log('clicked!')}>Just a button</Button>
      <Button primary>A primary button</Button>
      <Game />
    </Container>
  );
}

export default App;
