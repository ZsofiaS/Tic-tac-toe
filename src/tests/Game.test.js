import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Game, { ResultTitle } from '../components/Game';
import { Board } from '../components/Board';

let gameComponent;

beforeEach(() => {
  gameComponent = shallow(<Game />);
});

describe('renders Game component', () => {
  it('renders without crashing', () => {
    shallow(<Game />);
  });
  it('should render a p element', () => {
    expect(gameComponent.find(ResultTitle)).toHaveLength(1);
  });
  it('should render a board component', () => {
    expect(gameComponent.find(Board)).toHaveLength(1);
  });
  it('renders Game correctly', () => {
    expect(toJson(gameComponent)).toMatchSnapshot();
  });
});
