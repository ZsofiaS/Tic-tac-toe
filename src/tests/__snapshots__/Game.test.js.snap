// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Game component renders Game correctly 1`] = `
<Fragment>
  <p>
    X
    's turn
  </p>
  <Board
    current="X"
    declareWinner={[Function]}
    restart={[Function]}
    updateHandler={[Function]}
  />
</Fragment>
`;
